src/random.cpp

```diff
@@ -43,15 +43,21 @@ static void RandFailure()
 
 static inline int64_t GetPerformanceCounter()
 {
-    int64_t nCounter = 0;
-#ifdef WIN32
-    QueryPerformanceCounter((LARGE_INTEGER*)&nCounter);
+#if defined(_MSC_VER)
+    return __rdtsc();
+#elif defined(__i386__)
+    uint64_t r;
+    __asm__ volatile ("rdtsc" : "=A"(r));
+    return r;
+#elif defined(__x86_64__) || defined(__amd64__)
+    uint64_t r1, r2;
+    __asm__ volatile ("rdtsc" : "=a"(r1), "=d"(r2));
+    return (r2 << 32) | r1;
```

Comment:
> If resuming from hibernation, might this counter be, or close to, zero?

I don't think that's a concern. It's still a fine entropy source. This thing counts in clock cycles (~nanoseconds and less, usually). If an attacker knows how many nanoseconds ago your last wakeup from hibernation was, you probably have bigger problems.